src/hub.ts

```diff
@@ -50,22 +51,37 @@ export class SignalRHub implements ISignalRHub {
     private _connection: SignalR.Hub.Connection | undefined;
     private _proxy: SignalR.Hub.Proxy | undefined;
     private _startSubject = new Subject<void>();
+    private _stopSubject = new Subject<void>();
     private _stateSubject = new Subject<string>();
     private _errorSubject = new Subject<SignalR.ConnectionError>();
     private _subjects: { [name: string]: Subject<any> } = {};
 
     start$: Observable<void>;
+    stop$: Observable<void>;
     state$: Observable<string>;
     error$: Observable<SignalR.ConnectionError>;
 
     options?: SignalR.ConnectionOptions;
 
     constructor(public hubName: string, public url?: string) {
         this.start$ = this._startSubject.asObservable();
+        this.stop$ = this._stopSubject.asObservable();
         this.state$ = this._stateSubject.asObservable();
         this.error$ = this._errorSubject.asObservable();
     }
 
+    stop(async?: boolean, notifyServer?: boolean): Observable<void> {
+        if (this._connection) {
+            try{
+                this._connection.stop(async, notifyServer);
+                this._stopSubject.next();
+            } catch (error) {
+                this._stopSubject.error(error);
+            }
+        }
+        return this._stopSubject.asObservable();
```

Comment:
You might need to add a `setTimeout` around the try/catch because if you subscribe directly to the returned observable, you won't receive the next/error result.

Reply:
Fixed thanks